manager/controllers/app/fybrikapplication_conditions.go

```diff
@@ -46,33 +44,30 @@ func resetAssetState(application *api.FybrikApplication, assetID string) {
 	application.Status.AssetStates[assetID] = api.AssetState{Conditions: conditions}
 }
 
-func setErrorCondition(log zerolog.Logger, application *api.FybrikApplication, assetID string, msg string) {
+func setErrorCondition(appContext ApplicationContext, assetID string, msg string) {
 	errMsg := "An error was received for asset " + assetID
 	errMsg += " . If the error persists, please contact an operator."
 	errMsg += "Error description: " + msg
-	application.Status.AssetStates[assetID].Conditions[ErrorConditionIndex] = api.Condition{
+	appContext.Application.Status.AssetStates[assetID].Conditions[ErrorConditionIndex] = api.Condition{
 		Type:    api.ErrorCondition,
 		Status:  corev1.ConditionTrue,
 		Message: errMsg}
-	log.Error().Bool(logging.FORUSER, true).Bool(logging.AUDIT, true).
-		Str(utils.FybrikAppUUID, utils.GetFybrikApplicationUUID(application)).
-		Str(logging.DATASETID, assetID).Msg("Setting error condition: " + errMsg)
+	appContext.Log.Error().Bool(logging.FORUSER, true).Bool(logging.AUDIT, true).
+		Str(logging.DATASETID, assetID).Msgf("Setting error condition: %s", errMsg)
```

Comment:
Please do appContext.Log.Error().Err(errMsg) instead of appending errMsg to the message string

Reply:
And add the UUID

Reply:
@simanadler 
uuid is passed to all relevant methods/structures via applicationContext.Log.
It is constructed in FybrikApplication controller immediately after retrieving an object to reconcile:

        log := sublog.With().Str(utils.FybrikAppUUID, uuid).Logger()
	// Log the fybrikapplication
	logging.LogStructure("fybrikapplication", application, log, true, true)
	applicationContext := ApplicationContext{Log: log, Application: application, UUID: uuid}